src/NBitcoin/ChainedHeader.cs

```diff
@@ -80,6 +143,7 @@ public ChainedHeader(BlockHeader header, uint256 headerHash, int height)
             this.Header = header ?? throw new ArgumentNullException("header");
             this.Height = height;
             this.HashBlock = headerHash;
+            this.Next = new List<ChainedHeader>();
```

Comment:
I guess it will be a good optimization to initialize a list with capacity of 1 since in most of the cases ChainedHeader will have a single Next pointer. 
Same suggestion for the line 116